Add file with information on making a release, closes #237 (#332)
unitaryfund/mitiq: docs/source/release.rst

Generate the html tree and the pdf file for the documentation
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
- To create the html structure

.. code-block:: shell-session

	$ make html

and for the pdf,

.. code-block:: shell-session

	$ make latexpdf

Since the ``docs/build`` folder is not kept track of, copy the pdf file
with the documentation from ``docs/build/latex`` to the ``docs/pdf`` folder, saving it as ``mitiq.pdf``, thus replacing the previous version.
Add a copy of the pdf file by naming it according to the release version with major and minor, e.g., ``mitiq-0.1.pdf`` in the same folder.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Create a distribution locally
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Go to your local directory from the Terminal, e.g., ``github/mitiq/`` and there run

.. code-block:: shell-session

	$ python setup.py sdist bdist_wheel

This will create a source distribution and a built distribution with wheel. This should create a ``build/`` and ``sdist/`` folder.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Release the new version on Github
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.. note::

> 	You need to be a write access of the ``mitiq``'s' Github repository to make a new release.

Reviewer: ```suggestion
	You need to have write access to the ``mitiq``'s' Github repository to make a new release.
```

Make a new release on Github from the website, at https://github.com/unitaryfund/mitiq/releases.

	- Choose a tag and add information on the release using the same style used for previous releases.

	- Make sure that a branch with the version number has been created and is referenced for the given release.

	- Github will create compressed files with the repository. Upload the ``mitiq.pdf`` file and add the locally generated distribution and the Python wheels.

	- Add a brief text with a description of the version release.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Release the new versionon TestPyPI

Reviewer: ```suggestion
Release the new versionon on TestPyPI
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Before uploading the package on PyPI, since that action cannot be undone, it is good practice to upload it on the test channel TestPyPI.

.. note::
	You need to be a registered user on TestPyPI and a mantainer of the ``mitiq`` project in order to be able to upload the package.

- Upload the package. In order to upload it, you need to have ``twine``, which can be installed with ``pip install twine``. Go to the ``mitiq`` directory, after having created the source distribution version ``sdist``, and simply type

.. code-block:: shell-session

	$ twine upload --repository testpypi dist/*

You can then check at https://test.pypi.org/project/mitiq that the library is correctly uploaded.

- In order to check that the distribution runs correctly, set up a new (conda) environment and try to install the library, for example for version 0.1a1 this is done with:

.. code-block:: shell-session

> 	$ pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.python.org/simple/ mitiq==0.1a1

Reviewer 2: Change `mitiq==0.1a1` to the latest version.

Author: @rmlarose this was intentional. I wanted to give an example of how to pinpoint to a version. In the text above, it says: "for example for version 0.1a1 this is done with:...". If you think it is better to pinpoint to latest, feel free to suggest an edit. We want to make sure this is the case even if the latest version is not stable, or if the latest did not really update.

Reviewer 2: Ah I missed the text, sorry. This is fine with me.
